lib/src/query/query.dart

```diff
@@ -0,0 +1,571 @@
+    switch (_type) {
+      integer:
+      case OBXPropertyType.Int: // 4 bytes
+        switch (_op) {
+          case ConditionOp.eq:
+            return _op1(
+                cBuilder, propertyId, bindings.obx_qb_int_equal);
+          case ConditionOp.not_eq:
+            return _op1(
+                cBuilder, propertyId, bindings.obx_qb_int_not_equal);
+          case ConditionOp.gt:
+            return _op1(
+                cBuilder, propertyId, bindings.obx_qb_int_greater);
+          case ConditionOp.lt:
+            return _op1(
+                cBuilder, propertyId, bindings.obx_qb_int_less);
+          case ConditionOp.tween:
+            return bindings.obx_qb_int_between(
+                cBuilder, propertyId, _value, _value2);
+        }
+        break;
+      case OBXPropertyType.Long:
+        continue integer;
+      case OBXPropertyType.Byte:
+        continue integer;
+      case OBXPropertyType.Short:
+        continue integer;
+      case OBXPropertyType.Char:
+        continue integer;
+      case OBXPropertyType.Bool:
+        continue integer;
+    }
+
+    switch (_op) {
+      case ConditionOp.inside:
+        {
+          switch (_type) {
+            case OBXPropertyType.Int:
+              return _opList32(cBuilder, propertyId, bindings.obx_qb_int32_in);
+            case OBXPropertyType.Long:
+              return _opList64(cBuilder, propertyId, bindings.obx_qb_int64_in);
+          }
+          break;
+        }
+      case ConditionOp.not_in:
+        {
+          switch (_type) {
+            case OBXPropertyType.Int:
+              return _opList32(cBuilder, propertyId, bindings.obx_qb_int32_not_in);
+            case OBXPropertyType.Long:
+              return _opList64(cBuilder, propertyId, bindings.obx_qb_int64_not_in);
+          }
+          break;
+        }
+      default:
+        throw Exception("Unsupported operation ${_op.toString()}");
+    }
+  }
+}
+
+class DoubleCondition extends Condition<double> {
+  DoubleCondition(ConditionOp op, int type, double value, double value2)
+      : super(op, type, value, value2);
+
+  int _op1(Pointer<Void> qbPtr, int propertyId, obx_qb_cond_operator_1_dart_t<double> func) {
+    return func(qbPtr, propertyId, _value);
+  }
+
+  int apply(Pointer<Void> cBuilder, int propertyId) {
+    switch (_op) {
+      case ConditionOp.gt:
+        return _op1(
+        cBuilder, propertyId, bindings.obx_qb_double_greater);
+      case ConditionOp.lt:
+        return _op1(cBuilder, propertyId, bindings.obx_qb_double_less);
+      case ConditionOp.tween:
+        return bindings.obx_qb_double_between(
+            cBuilder, propertyId, _value, _value2);
+      default:
+        throw Exception("Unsupported operation ${_op.toString()}");
+    }
+  }
+}
+
+/**
+ * The first element of the chain
+ * contains the enum representation of
+ * the to-be-constructed query builder.
+ * This design allows nested chains inside
+ * the chain.
+ */
+class QueryCondition {
```

Comment:
I believe that `any`/`all` conditions should be group together as many conditions as possible, if they're on the same logical level anyway.

Reply:
Hm, I have to learn to read better. For single conditions, this makes sense. I'll take a look again.

Reply:
I prefer composition to inheritance in this case, because the all/any operators just take a builder and an array of conditions. No `entityId`, no `propertyId`.

Reply:
Are you implying that you want to get rid of `build` and directly create the condition objects? I believe that the _swift_ project does that, they just pass a closure.

Reply:
As long as the algorithm efficiently creates trees of conditions, I'm fine with it. You've collapsed some classes together, it looks more cohesive like this.

I especially like this:
```dart
final list = [this, ...rest];
```
I wasn't aware of this syntax.

Merged.